src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -38,15 +40,54 @@ CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
                 break;
         }
     }
-    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+
     ReturnErrorOnFailure(reader.ExitContainer(outer));
     return CHIP_NO_ERROR;
 }
 
 } // namespace {{asType label}}
 {{/zcl_structs}}
-} // namespace {{asUpperCamelCase name}}
 
+namespace Commands {
+{{#zcl_commands}}
+namespace {{asUpperCamelCase name}} {
+CHIP_ERROR Type::Encode(TLV::TLVWriter &writer, uint64_t tag) const{
+    TLV::TLVType outer;
+    ReturnErrorOnFailure(writer.StartContainer(tag, TLV::kTLVType_Structure, outer));
+    {{#zcl_command_arguments}}
+    ReturnErrorOnFailure(DataModel::Encode(writer, TLV::ContextTag(k{{asUpperCamelCase label}}FieldId), {{asLowerCamelCase label}}));
+    {{/zcl_command_arguments}}
+    ReturnErrorOnFailure(writer.EndContainer(outer));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
+    CHIP_ERROR err;
+    TLV::TLVType outer;
+    VerifyOrReturnError(chip::TLV::kTLVType_Structure == reader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
+    err = reader.EnterContainer(outer);
+    ReturnErrorOnFailure(err);
+    while ((err = reader.Next()) == CHIP_NO_ERROR) {
+        VerifyOrReturnError(chip::TLV::IsContextTag(reader.GetTag()), CHIP_ERROR_INVALID_TLV_TAG);
+        switch (chip::TLV::TagNumFromTag(reader.GetTag()))
+        {
+            {{#zcl_command_arguments}}
+            case k{{asUpperCamelCase label}}FieldId:
+                ReturnErrorOnFailure(DataModel::Decode(reader, {{asLowerCamelCase label}}));
+                break;
+                {{/zcl_command_arguments}}
+            default:
+                break;
+        }
+    }
+
+    ReturnErrorOnFailure(reader.ExitContainer(outer));
```

Comment:
This line was there before in line 43:

```
VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
```

Why was that removed?

Reply:
From previous boris's comment, This will fail out if there are optional arguments we don't know about (i.e. from a newer spec version). That does not seem right. We should just ignore them. Please add a test.

Looks to me like existing struct codegen has the same problem....

Reply:
Done, thanks